### PR TITLE
.deb package is now built during Maven build.

### PR DESCRIPTION
**Pull request by gehel**
All credit for this commit goes to @praste. This is a rewrite of https://github.com/jmxtrans/jmxtrans/pull/228.

Discussion around this move to Maven can also be found in #166.

Debian control files have been moved to /src/deb to follow standard Maven structure more closely. Build script should package and upload .deb file to Sonatype OSS (untested).
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23issuecomment-76175025%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23issuecomment-76255409%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23issuecomment-77250666%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23issuecomment-77273092%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23discussion_r25825690%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23discussion_r25873672%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23discussion_r25873895%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23issuecomment-77394158%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23discussion_r25876406%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23discussion_r25890474%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23issuecomment-77426663%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23discussion_r25907040%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23issuecomment-79042742%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23issuecomment-80368878%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23discussion_r26449473%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23issuecomment-81159787%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23issuecomment-76175025%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Looks%20good.%20Thanks%20a%20lot%20for%20taking%20this%20up.%20%22%2C%20%22created_at%22%3A%20%222015-02-26T13%3A09%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6951770%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/praste%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-02-26T19%3A51%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22Seems%20that%20%60jmxtrans-all.jar%60%20is%20referenced%20with%20a%20relative%20path%20from%20%60/usr/share/jmxtrans/bin/jmxtrans.sh%60%20and%20as%20such%20it%20is%20not%20found.%20I%27m%20not%20sure%20if%20I%20should%20try%20to%20fix%20the%20%60jmxtrans.sh%60%20script%2C%20or%20move%20everything%20to%20Java%20Service%20Wrapper%20%28like%20the%20RPM%29.%20Note%20that%20moving%20to%20java%20service%20wrapper%20will%20introduce%20incompatibilities%20in%20term%20of%20configuration%20format.%22%2C%20%22created_at%22%3A%20%222015-03-04T21%3A18%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20think%20issue%20about%20%60jmxtrans.sh%60%20referring%20to%20%60jmxtrans-all.jar%60%20is%20in%20%60dist%60%20as%20well.%20I%20am%20not%20sure%20how%20others%20addressing%20it%2C%20but%20I%20do%20tinker%20with%20the%20%60jmxtrans.sh%60%20%20to%20fix%20many%20variables.%5Cr%5Cn%5Cr%5CnCan%20adding%20%60JAR_FILE%60%20variable%20point%20to%20absolute%20path%20of%20%60jmxtrans-all.jar%60%20to%20%60CONFIG%60%20file%20help%3F%22%2C%20%22created_at%22%3A%20%222015-03-04T23%3A23%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6951770%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/praste%22%7D%7D%2C%20%7B%22body%22%3A%20%22Absolute%20reference%20added%20to%20jar%20file.%20Fixes%20https%3A//github.com/jmxtrans/jmxtrans/pull/231%23issuecomment-77273092%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-03-05T16%3A17%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22To%20remove%20can%20you%20try%20%60dpkg%20-r%60%22%2C%20%22created_at%22%3A%20%222015-03-05T18%3A58%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6951770%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/praste%22%7D%7D%2C%20%7B%22body%22%3A%20%22Can%20this%20be%20merged%20to%20%60master%60%22%2C%20%22created_at%22%3A%20%222015-03-13T15%3A24%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6951770%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/praste%22%7D%7D%2C%20%7B%22body%22%3A%20%22No%2C%20this%20cannot%20be%20merged.%20Package%20does%20not%20install%20correctly%20%28service%20does%20not%20start%29%20and%20does%20not%20uninstall%20correctly.%20I%20will%20try%20to%20find%20some%20time%20to%20correct%20those%202%20errors%2C%20but%20no%20promsie%20on%20the%20deadline%2C%20sorry.%20As%20always%2C%20help%20is%20welcomed.%22%2C%20%22created_at%22%3A%20%222015-03-14T11%3A49%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22Corrected%20startup%20script%2C%20it%20now%20starts%20correctly%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-03-15T17%3A05%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20327243c5ceb0644edee3d0d350b880999ae6fdad%20src/deb/control/postinst%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23discussion_r25825690%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22can%20you%20please%20verify%20if%20this%20location%20is%20in%20sync%20with%20one%20on%20line%2093%20and%20default%20conf%20location%20%60jmxtrans.sh%60%20looks%20for%22%2C%20%22created_at%22%3A%20%222015-03-04T23%3A27%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6951770%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/praste%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yes%2C%20it%20is%20configured%20correctly.%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-03-05T15%3A52%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/deb/control/postinst%3AL6-13%22%7D%2C%20%22Pull%20327243c5ceb0644edee3d0d350b880999ae6fdad%20pom.xml%2036%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/231%23discussion_r25873895%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60apt-get%20remove%60%20seem%20to%20leave%20everything%20in%20place.%5Cr%5CnNot%20really%20sure%20how%20the%20archive%20type%20works%2C%20but%20I%20suspect%20a%20cause%20around%20here.%20More%20investigation%20required.%22%2C%20%22created_at%22%3A%20%222015-03-05T15%3A55%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22.deb%20deploys%20more%20or%20less%20correctly%20%28at%20least%20JmxTrans%20is%20starting%20without%20error%29.%20%60apt-get%20remove%60%20still%20does%20not%20seem%20to%20work.%5Cr%5Cn%5Cr%5CnAs%20this%20is%20already%20a%20working%20increment%20over%20the%20current%20situation%20%28no%20.deb%20created%20at%20all%29%2C%20I%20would%20tend%20to%20merge%20it%20as-is.%20But%20I%27d%20like%20to%20have%20your%20feedback%20before.%22%2C%20%22created_at%22%3A%20%222015-03-05T16%3A21%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22http%3A//askubuntu.com/questions/231562/what-is-the-difference-between-apt-get-purge-and-apt-get-remove%22%2C%20%22created_at%22%3A%20%222015-03-05T18%3A55%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/85355%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lookfirst%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yep%2C%20I%27m%20aware%20of%20the%20difference.%20But%20%60apt-get%20remove%60%20should%20remove%20the%20binaries%20%28jmxtrans-all.jar%20for%20example%29%20and%20it%20does%20not%20at%20this%20point.%20Bad%20...%22%2C%20%22created_at%22%3A%20%222015-03-05T22%3A12%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22Seems%20I%27ve%20been%20smoking.%20Cannot%20reproduce%20the%20issue.%5Cr%5Cn%5Cr%5Cn%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-03-15T17%3A04%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20pom.xml%3AL1157-1212%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/jmxtrans/jmxtrans/pull/231#issuecomment-76175025'>General Comment</a></b>
- <a href='https://github.com/praste'></a> Looks good. Thanks a lot for taking this up.
- <a href='https://github.com/gehel'></a> :+1:
- <a href='https://github.com/gehel'></a> Seems that `jmxtrans-all.jar` is referenced with a relative path from `/usr/share/jmxtrans/bin/jmxtrans.sh` and as such it is not found. I'm not sure if I should try to fix the `jmxtrans.sh` script, or move everything to Java Service Wrapper (like the RPM). Note that moving to java service wrapper will introduce incompatibilities in term of configuration format.
- <a href='https://github.com/praste'></a> I think issue about `jmxtrans.sh` referring to `jmxtrans-all.jar` is in `dist` as well. I am not sure how others addressing it, but I do tinker with the `jmxtrans.sh`  to fix many variables.
Can adding `JAR_FILE` variable point to absolute path of `jmxtrans-all.jar` to `CONFIG` file help?
- <a href='https://github.com/gehel'></a> Absolute reference added to jar file. Fixes https://github.com/jmxtrans/jmxtrans/pull/231#issuecomment-77273092 :+1:
- <a href='https://github.com/praste'></a> To remove can you try `dpkg -r`
- <a href='https://github.com/praste'></a> Can this be merged to `master`
- <a href='https://github.com/gehel'></a> No, this cannot be merged. Package does not install correctly (service does not start) and does not uninstall correctly. I will try to find some time to correct those 2 errors, but no promsie on the deadline, sorry. As always, help is welcomed.
- <a href='https://github.com/gehel'></a> Corrected startup script, it now starts correctly :+1:
- [x] <a href='#crh-comment-Pull 327243c5ceb0644edee3d0d350b880999ae6fdad src/deb/control/postinst 5'></a> &nbsp;<b><a href='https://github.com/jmxtrans/jmxtrans/pull/231#discussion_r25825690'>File: src/deb/control/postinst:L6-13</a></b>
- <a href='https://github.com/praste'></a> can you please verify if this location is in sync with one on line 93 and default conf location `jmxtrans.sh` looks for
- <a href='https://github.com/gehel'></a> Yes, it is configured correctly. :+1:
- [x] <a href='#crh-comment-Pull 327243c5ceb0644edee3d0d350b880999ae6fdad pom.xml 36'></a> &nbsp;<b><a href='https://github.com/jmxtrans/jmxtrans/pull/231#discussion_r25873895'>File: pom.xml:L1157-1212</a></b>
- <a href='https://github.com/gehel'></a> `apt-get remove` seem to leave everything in place.
Not really sure how the archive type works, but I suspect a cause around here. More investigation required.
- <a href='https://github.com/gehel'></a> .deb deploys more or less correctly (at least JmxTrans is starting without error). `apt-get remove` still does not seem to work.
As this is already a working increment over the current situation (no .deb created at all), I would tend to merge it as-is. But I'd like to have your feedback before.
- <a href='https://github.com/lookfirst'></a> http://askubuntu.com/questions/231562/what-is-the-difference-between-apt-get-purge-and-apt-get-remove
- <a href='https://github.com/gehel'></a> Yep, I'm aware of the difference. But `apt-get remove` should remove the binaries (jmxtrans-all.jar for example) and it does not at this point. Bad ...
- <a href='https://github.com/gehel'></a> Seems I've been smoking. Cannot reproduce the issue.
:+1:

<a href='https://www.codereviewhub.com/jmxtrans/jmxtrans/pull/231?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/jmxtrans/jmxtrans/pull/231?approve=1'></a>&nbsp;<a href='https://github.com/jmxtrans/jmxtrans/pull/231'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by gehel** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/jmxtrans/jmxtrans/pull/231'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by praste** [comment]
Looks good. Thanks a lot for taking this up.

**Comment by gehel** [comment]
:+1:

**Comment by gehel** [comment]
Seems that `jmxtrans-all.jar` is referenced with a relative path from `/usr/share/jmxtrans/bin/jmxtrans.sh` and as such it is not found. I'm not sure if I should try to fix the `jmxtrans.sh` script, or move everything to Java Service Wrapper (like the RPM). Note that moving to java service wrapper will introduce incompatibilities in term of configuration format.

**Comment by praste** [comment]
I think issue about `jmxtrans.sh` referring to `jmxtrans-all.jar` is in `dist` as well. I am not sure how others addressing it, but I do tinker with the `jmxtrans.sh`  to fix many variables.

Can adding `JAR_FILE` variable point to absolute path of `jmxtrans-all.jar` to `CONFIG` file help?

File: src/deb/control/postinst [hunk 1]
```diff
@@ -6,7 +6,7 @@ set -e
 . /usr/share/debconf/confmodule
 
 # defaults file
-CONFIG="/etc/default/@PACKAGE@"
+CONFIG="/etc/default/[[name]]"
```
Comment (praste) [inline src/deb/control/postinst]: can you please verify if this location is in sync with one on line 93 and default conf location `jmxtrans.sh` looks for

Comment (gehel) [inline src/deb/control/postinst]: Yes, it is configured correctly. :+1:

File: pom.xml [hunk 2]
```diff
@@ -1155,6 +1157,55 @@
...
+							<execution>
+								<goals>
+									<goal>jdeb</goal>
+								</goals>
+								<configuration>
+									<skipPOMs>true</skipPOMs>
+									<deb>${project.build.directory}/${project.artifactId}-${project.version}.deb</deb>
+									<dataSet>
+										<data>
+											<src>${project.build.directory}/jmxtrans-${project.version}-dist.tar.gz</src>
```
Comment (gehel) [inline pom.xml]: `apt-get remove` seem to leave everything in place.
Not really sure how the archive type works, but I suspect a cause around here. More investigation required.

**Comment by gehel** [comment]
Absolute reference added to jar file. Fixes https://github.com/jmxtrans/jmxtrans/pull/231#issuecomment-77273092 :+1:

File: pom.xml (hunk 2 above)
Comment (gehel) [inline pom.xml]: .deb deploys more or less correctly (at least JmxTrans is starting without error). `apt-get remove` still does not seem to work.

As this is already a working increment over the current situation (no .deb created at all), I would tend to merge it as-is. But I'd like to have your feedback before.

Comment (lookfirst) [inline pom.xml]: http://askubuntu.com/questions/231562/what-is-the-difference-between-apt-get-purge-and-apt-get-remove

**Comment by praste** [comment]
To remove can you try `dpkg -r`

File: pom.xml (hunk 2 above)
Comment (gehel) [inline pom.xml]: Yep, I'm aware of the difference. But `apt-get remove` should remove the binaries (jmxtrans-all.jar for example) and it does not at this point. Bad ...

**Comment by praste** [comment]
Can this be merged to `master`

**Comment by gehel** [comment]
No, this cannot be merged. Package does not install correctly (service does not start) and does not uninstall correctly. I will try to find some time to correct those 2 errors, but no promsie on the deadline, sorry. As always, help is welcomed.

File: pom.xml (hunk 2 above)
Comment (gehel) [inline pom.xml]: Seems I've been smoking. Cannot reproduce the issue.

:+1:

**Comment by gehel** [comment]
Corrected startup script, it now starts correctly :+1:

**Pull request merged**